### PR TITLE
fix: add a deprecation warning to `jx install`/`jx upgrade platform`

### PR DESCRIPTION
**Pull request by jstrachan**
to give folks a hint they should switch to `jx boot`

Signed-off-by: James Strachan <james.strachan@gmail.com>

### CODE REVIEW

**Comment by garethjevans** [comment]
@jstrachan should we add this to `jx create cluster <xxx>` too? (maybe if the --skip-installation flag is not set)

**Comment by jstrachan** [comment]
@garethjevans good call! I did wonder about doing that - but I wondered if `jx create cluster` without the `--skip-installation` would invoke install and so get the warning anyway. I've not tested that though ;)

File: pkg/cmd/opts/deprecations.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package opts
+
+import (
+	"github.com/jenkins-x/jx/pkg/log"
+	"github.com/jenkins-x/jx/pkg/util"
+)
+
+// LogInstallDeprecated logs a warning message about deprecation
+func LogInstallDeprecated() {
```
Comment (ccojocar) [inline pkg/cmd/opts/deprecations.go]: I would make this function more generic, that other commands can use it. e.g. provide the `recommended command` as an argument and also the link to the docs.

Comment (ccojocar) [reply pkg/cmd/opts/deprecations.go]: also `Cobra Command` seems to have a `Deprecated` flag which can be set on the command.

```
type Command struct {
	// Deprecated defines, if this command is deprecated and should print this string when used.
	Deprecated string
```

which prints this message before the command is executed:

```Go
func (c *Command) execute(a []string) (err error) {
	if c == nil {
		return fmt.Errorf("Called Execute() on a nil Command")
	}

	if len(c.Deprecated) > 0 {
		c.Printf("Command %q is deprecated, %s\n", c.Name(), c.Deprecated)
	}
```

I am wondering if we can leverage this instead.

**Comment by ccojocar** [comment]
@jstrachan We can close this PR. I implemented a more generic mechanics to mark commands as deprecated. Please have a look over https://github.com/jenkins-x/jx/pull/6063. Thanks!

**Comment by jstrachan** [comment]
@ccojocar awesome thanks!

**Pull request closed without merging**